src/Microsoft.Azure.SignalR.Protocols/SequenceOfT.cs

```diff
@@ -0,0 +1,465 @@
+﻿// Copyright (c) Andrew Arnott. All rights reserved.
+// Licensed under the MIT license. See LICENSE.txt file in the project root for full license information.
+
+#nullable enable
+
+namespace Nerdbank.Streams
+{
+    using System;
+    using System.Buffers;
+    using System.Collections.Generic;
+    using System.ComponentModel;
+    using System.Diagnostics;
+    using System.Reflection;
+    using System.Runtime.CompilerServices;
+    using System.Runtime.InteropServices;
+    using Microsoft;
+
+    /// <summary>
+    /// Manages a sequence of elements, readily castable as a <see cref="ReadOnlySequence{T}"/>.
+    /// </summary>
+    /// <typeparam name="T">The type of element stored by the sequence.</typeparam>
+    /// <remarks>
+    /// Instance members are not thread-safe.
+    /// </remarks>
+    [DebuggerDisplay("{" + nameof(DebuggerDisplay) + ",nq}")]
+    public class Sequence<T> : IBufferWriter<T>, IDisposable

```

Comment:
This will need to be made internal if you stick with this build-from-source design.

Reply:
I would not have guessed this one ^^

Reply:
This one would conflict with the public type in the Nerdbank.Streams package.

Reply:
Yeah, to be fair i'm not yet confident with `build from source` as this is something I never used before (for obvious reason).
It seems completely logic and obvious just like the `#if PRIVATE_BUILD` and `$(DefinedConstant)`
it's just not a reflex yet.